src/main/java/one/profiler/AsyncProfilerLoader.java

```diff
@@ -124,16 +122,31 @@ public static void deleteExtractionDirectory() throws IOException {
   /** Returns the directory used for storing the extracted libraries, binaries and JARs */
   public static Path getExtractionDirectory() throws IOException {
     if (extractionDir == null) {
-      extractionDir =
-          Paths.get(
-              ProjectDirectories.from("me", "bechberger", "ap-loader-" + getVersion()).dataDir);
+      extractionDir = Paths.get(getApplicationsDir(), "me.bechberger.ap-loader", getVersion());
       if (Files.notExists(extractionDir)) {
         Files.createDirectories(extractionDir);
       }
     }
     return extractionDir;
   }
 
+  /**
+   * Returns directory where applications places their files. Specific to operating system
+   */
+  private static String getApplicationsDir() {
+    String os = System.getProperty("os.name").toLowerCase();
+    if (os.contains("linux")) {
+      String xdgDataHome = System.getenv("XDG_DATA_HOME");
+      if (xdgDataHome != null && !xdgDataHome.isEmpty()) {
+        return xdgDataHome;
+      }
+      return Paths.get(System.getProperty("user.home"), ".local", "share").toString();
+    } else if (os.contains("mac")) {
```

Comment:
Please use the `os.startsWith` approach from `getLibrarySuffix`, so it is consistent.